Src/FluentAssertions/Formatting/DefaultValueFormatter.cs

```diff
@@ -40,20 +37,24 @@ public string Format(object value, FormattingContext context, FormatChild format
 
             if (HasDefaultToStringImplementation(value))
             {
-                return prefix + GetTypeAndPublicPropertyValues(value, context, formatChild);
+                return prefix + GetTypeAndMemberValues(value, context, formatChild);
             }
 
             return prefix + value;
         }
 
+        protected virtual IEnumerable<SelectedMemberInfo> GetMembers(Type type) => type.GetNonPrivateMembers();
+
+        protected virtual string GetTypeName(Type type) => type.FullName;
```

Comment:
Why do you need this in sub-classes?

Reply:
At work I made a custom formatter that only returned `type.Name` as all types where from the same namespace.
It cluttered up the message as the subject was a collection of quite nested objects.

So it went from
```
A.Very.Long.Namespace.Class1
{
   Prop1 = A.Very.Long.Namespace.Class2
   {
      ...
   }
}
```
into
```
Class1
{
   Prop1 = Class2
   {
      ...
   }
}```

But I agree it might be out of scope for most people who would want to extend this class.

Maybe it just needs a better name?
`FormatType`
`FormatTypeName`

Reply:
Ah, maybe `TypeDisplayName`

Reply:
Eureka 🎉 
DisplayName, that's the right word!
Naming is hard.